### PR TITLE
#55 Feature: Users can reset password

### PR DESCRIPTION
**Pull request by faid-terence**
#### What does this PR do?
This pull request adds a new feature to the application: a reset password page. The reset password page allows users to reset their passwords in case they forget or want to change them.
#### Description of Task to be completed?
In case the users password is forgotten, functionality should exists to reset the password via an associated and email account.

#### How should this be manually tested?
```
git clone https://github.com/atlp-rwanda/ecommerce-app-predators-fe.git
cd ecommerce-app-predators-fe
git checkout branch ft-reset-user-password
npm i
npm run dev
http://localhost:5173/reset-password-page
```

#### What are the relevant pivotal tracker/Trello stories?
Trello [Link](https://trello.com/c/SXhRxTFm/55-users-should-be-presented-with-a-reset-password-link) (#-55)

### CODE REVIEW

**Comment by faid-terence** [comment]
[Users should be presented with a reset password link](https://trello.com/c/SXhRxTFm/55-users-should-be-presented-with-a-reset-password-link)

**Comment by faid-terence** [comment]
[Users should be presented with a reset password link](https://trello.com/c/SXhRxTFm/55-users-should-be-presented-with-a-reset-password-link)

**Review by Muhet — APPROVED** [review]
GOOD JOB

**Review by yobuir — APPROVED** [review]

File: src/pages/PasswordConfirminationPage..tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+    if (newPassword !== confirmPassword) {
+      // Display an error message for password mismatch
+      // You can use a toast library or any other way to show the message
+      toast.error("Please Enter Same Password")
+    }
+    dispatch(updatePassword(resetData) as any)
```
Comment (chanceDira) [inline src/pages/PasswordConfirminationPage..tsx:33]: Let's not use any keyword, create an interface for resetData

File: src/pages/RegisterPage.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
+import { SetStateAction, useState, } from 'react';
+import { useDispatch, useSelector } from "react-redux";
+import { ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { registerUser } from '../redux/action/UserAction';
+import store from '../../src/assets/store.svg'
+const RegisterPage = () => {
+    const dispatch = useDispatch();
+    const registrationStatus = useSelector((state: any) => state.user.status);
```
Comment (chanceDira) [inline src/pages/RegisterPage.tsx:9]: no "any" keyword

**Review by chanceDira — CHANGES_REQUESTED** [review]
Nice work @faid-terence  just few issues

**Pull request closed without merging**